### PR TITLE
chore(rest): correct cmake config and disambiguate C names

### PR DESCRIPTION
**Pull request by scotthart**
As part of the REST refactor effort, the extern "C" symbols names in the REST library needed to be different than the existing names in GCS.

Also fixes some of the REST libraries cmake configuration to follow existing conventions.

This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/8693)

### CODE REVIEW

File: google/cloud/config-rest.cmake.in [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+include(CMakeFindDependencyMacro)
+find_dependency(google_cloud_cpp_common)
+find_dependency(absl)
```
Comment (coryan) [inline google/cloud/config-rest.cmake.in:17]: Don't we need to find `CURL` also?

https://github.com/googleapis/google-cloud-cpp/blob/main/google/cloud/storage/config.cmake.in#L18

**Review by scotthart — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/8693)* status: 0 of 6 files reviewed, 1 unresolved discussion (waiting on @coryan)

___
*[google/cloud/config-rest.cmake.in, line 17 at r1](https://reviewable.io/reviews/googleapis/google-cloud-cpp/8693#-N--bdXDabkzObbfgdyg-r1-17:-N-OYiGa28sS3cg1kneM:bfvn68m) ([raw file](https://github.com/googleapis/google-cloud-cpp/blob/eaebbbcd859ebbc7c5f61c7c5a8dec836783a9fd/google/cloud/config-rest.cmake.in#L17)):*
<details><summary><i>Previously, coryan (Carlos O'Ryan) wrote…</i></summary><blockquote>

Don't we need to find `CURL` also?

https://github.com/googleapis/google-cloud-cpp/blob/main/google/cloud/storage/config.cmake.in#L18

</blockquote></details>

And a few others.

**Review by coryan — APPROVED** [review]
> Reviewed 5 of 6 files at r1, 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/8693)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @scotthart)

___
*[google/cloud/CMakeLists.txt, line 633 at r2](https://reviewable.io/reviews/googleapis/google-cloud-cpp/8693#-N-Oah4u2frutCleZV7z:-N-Oah4u2frutCleZV8-:b-y9silx) ([raw file](https://github.com/googleapis/google-cloud-cpp/blob/1920c9273352a842bb483b78d6738eba755fd4c2/google/cloud/CMakeLists.txt#L633)):*
> ```CMake
>     include(IncludeNlohmannJson)
>     include(FindCurlWithTargets)
>     find_package(OpenSSL REQUIRED)
> ```

FYI, this is also used by `google_cloud_cpp_common`.
___
*[google/cloud/CMakeLists.txt, line 694 at r2](https://reviewable.io/reviews/googleapis/google-cloud-cpp/8693#-N-OaZlm2pd1KrBRuJPK:-N-OaZlm2pd1KrBRuJPL:b-2x5q9n) ([raw file](https://github.com/googleapis/google-cloud-cpp/blob/1920c9273352a842bb483b78d6738eba755fd4c2/google/cloud/CMakeLists.txt#L694)):*
> ```CMake
>     target_link_libraries(
>         google_cloud_cpp_rest_internal
>         PUBLIC absl::span google-cloud-cpp::common CURL::libcurl
> ```

FYI, the list of `find_dependency()` calls should match what is used here.
___
*[google/cloud/config-rest.cmake.in, line 17 at r1](https://reviewable.io/reviews/googleapis/google-cloud-cpp/8693#-N--bdXDabkzObbfgdyg-r1-17:-N-OaSGz8BnNcjjI6zlI:b-lvj5tz) ([raw file](https://github.com/googleapis/google-cloud-cpp/blob/eaebbbcd859ebbc7c5f61c7c5a8dec836783a9fd/google/cloud/config-rest.cmake.in#L17)):*
<details><summary><i>Previously, scotthart (Scott Hart) wrote…</i></summary><blockquote>

And a few others.
</blockquote></details>

I do not think this library uses Crc32c.  Please remove.

**Review by scotthart — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/8693)* status: 5 of 6 files reviewed, all discussions resolved (waiting on @coryan)

___
*[google/cloud/CMakeLists.txt, line 694 at r2](https://reviewable.io/reviews/googleapis/google-cloud-cpp/8693#-N-OaZlm2pd1KrBRuJPK:-N-PMT9S-acQycQHzzQx:bahtfrr) ([raw file](https://github.com/googleapis/google-cloud-cpp/blob/1920c9273352a842bb483b78d6738eba755fd4c2/google/cloud/CMakeLists.txt#L694)):*
<details><summary><i>Previously, coryan (Carlos O'Ryan) wrote…</i></summary><blockquote>

FYI, the list of `find_dependency()` calls should match what is used here.
</blockquote></details>

Updated find_dependecy calls to match these linked libraries.
___
*[google/cloud/config-rest.cmake.in, line 17 at r1](https://reviewable.io/reviews/googleapis/google-cloud-cpp/8693#-N--bdXDabkzObbfgdyg-r1-17:-N-PMYDgA3KtLVgiE0qU:ba3dfob) ([raw file](https://github.com/googleapis/google-cloud-cpp/blob/eaebbbcd859ebbc7c5f61c7c5a8dec836783a9fd/google/cloud/config-rest.cmake.in#L17)):*
<details><summary><i>Previously, coryan (Carlos O'Ryan) wrote…</i></summary><blockquote>

I do not think this library uses Crc32c.  Please remove.
</blockquote></details>

Removed

**Pull request merged**